### DISCUSSION TITLE
Problems with Pasting Images into Outline Wiki

### DISCUSSION THREAD

**Discussion by MetaBliss** [outline/outline#5004] [Self Hosting]
I recently installed Outline Version 0.68.1, self-hosted on a Synology NAS with Docker installed. The other containers running with Outline are Redis, MinIO, Postgresql. 

I have Outline up and running fine. I can authenticate Outline with both Slack and KeyCloak. I can edit, modify the team name, and most other Outline features work (except exports which I still have to figure out why).

However, the problem is that I have a major issue with pasting or uploading images. When I try to copy an image online from any site online and paste it into a page I am editing, the image appears for a split second as a light shadow colour and then it dissapears and in its place a generic error message comes up. 

 The error message is "Upload Failed" which appears at the bottom left in Outline and after a few seconds it dissapears. I have taken a quick screen shot of it below. 

I have checked the logs for MinIO, Outline, Postgreql, and Redis. Nothing seems out of the ordinary. For reference, I've copied some of the most recent logs from each of the four containers below. And I've provided my Docker Compose below to for reference.

I also logged into MinIO admin panel to see if any images showed up in the bucket but nothing shows up in the bucket. I have made sure the name of the MinIO bucket is not the same as the container or domain name (hence why its weekee). 

<img width="542" alt="image" src="https://user-images.githubusercontent.com/84720384/223326545-0af6c047-5714-4835-ab3d-2e64e71d4d88.png">

Thank you SO much for your review and assistance with this problem I'm facing! It's driving me nuts. 

Outline Log:

> INF Processing attachments.create | name=attachments.create modelId=d5ac8112-70ca-4633-b3ce-55f5207f94e4 attempt=0 label=worker
INF WebhookProcessor running attachments.create | name=attachments.create modelId=d5ac8112-70ca-4633-b3ce-55f5207f94e4 label=worker
INF Processing attachments.create | name=attachments.create modelId=208201c8-4201-4618-af2d-c209b40139d5 attempt=0 label=worker
INF WebhookProcessor running attachments.create | name=attachments.create modelId=208201c8-4201-4618-af2d-c209b40139d5 label=worker
INF Processing attachments.create | name=attachments.create modelId=23f997bf-bd16-4691-a23e-d15783ce77dd attempt=0 label=worker
INF WebhookProcessor running attachments.create | name=attachments.create modelId=23f997bf-bd16-4691-a23e-d15783ce77dd label=worker
INF Processing attachments.create | name=attachments.create modelId=31f43d46-35fa-4009-82b9-e437bea196bf attempt=0 label=worker
INF WebhookProcessor running attachments.create | name=attachments.create modelId=31f43d46-35fa-4009-82b9-e437bea196bf label=worker
INF Processing attachments.create | name=attachments.create modelId=851f42f6-478d-4c93-b1a5-ea2422f73e8b attempt=0 label=worker
INF WebhookProcessor running attachments.create | name=attachments.create modelId=851f42f6-478d-4c93-b1a5-ea2422f73e8b label=worker
INF Persisting c5534657-d25d-4730-a7e1-92f81f4ff402, attributed to e3608480-5d81-49e1-a991-303a9f616f56 | label=multiplayer
INF Processing documents.update | name=documents.update attempt=0 label=worker
INF BacklinksProcessor running documents.update | name=documents.update label=worker
INF DebounceProcessor running documents.update | name=documents.update label=worker
INF WebhookProcessor running documents.update | name=documents.update label=worker

MinIO Log:
> MinIO` Object Storage Server
Copyright: 2015-2023 MinIO, Inc.
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Version: RELEASE.2023-02-27T18-10-45Z (go1.19.6 linux/amd64)
Status:         1 Online, 0 Offline. 
API: http://192.168.240.6:9000  http://127.0.0.1:9000 
Console: https://minio.mydomain.synology.me 
Documentation: https://min.io/docs/minio/linux/index.html
Warning: The standard parity is set to 0. This can lead to data loss.

Postgresql Log:

> PostgreSQL Database directory appears to contain a database; Skipping initialization
2023-03-06 23:56:24.051 UTC [1] LOG:  starting PostgreSQL 13.10 (Debian 13.10-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
2023-03-06 23:56:24.099 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-03-06 23:56:24.099 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2023-03-06 23:56:24.225 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-03-06 23:56:24.825 UTC [28] LOG:  database system was shut down at 2023-03-06 23:55:56 UTC
2023-03-06 23:56:25.902 UTC [1] LOG:  database system is ready to accept connections

Redis Log:

> 1:M 07 Mar 2023 05:08:38.674 * 100 changes in 300 seconds. Saving...
1:M 07 Mar 2023 05:08:38.674 * Background saving started by pid 43
43:C 07 Mar 2023 05:08:38.921 * DB saved on disk
43:C 07 Mar 2023 05:08:38.921 * Fork CoW for RDB: current 1 MB, peak 1 MB, average 0 MB
1:M 07 Mar 2023 05:08:38.977 * Background saving terminated with success
1:M 07 Mar 2023 05:13:39.007 * 100 changes in 300 seconds. Saving...
1:M 07 Mar 2023 05:13:39.008 * Background saving started by pid 44
44:C 07 Mar 2023 05:13:39.276 * DB saved on disk
44:C 07 Mar 2023 05:13:39.276 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 07 Mar 2023 05:13:39.309 * Background saving terminated with success
1:M 07 Mar 2023 05:28:11.324 * 100 changes in 300 seconds. Saving...
1:M 07 Mar 2023 05:28:11.324 * Background saving started by pid 45
45:C 07 Mar 2023 05:28:11.566 * DB saved on disk
45:C 07 Mar 2023 05:28:11.566 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 07 Mar 2023 05:28:11.626 * Background saving terminated with success

Below is my full Docker Compose Stack which I run on Portainer. All four services (Postgres, Redis, Outline and MinIO)

`version: "2"
services:
  outline_redis:
    image: redis
    restart: always
    container_name: outline_redis
    ports:
      - Port:6379
    networks:
      - outline-internal

  outline_postgres:
    image: postgres:13
    restart: always
    container_name: outline_postgres
    environment:
      - POSTGRES_PASSWORD=XXXXXXXXXXXXX
      - POSTGRES_USER=outline
      - POSTGRES_DB=outline
    volumes:
      - /volume1/docker/postgres:/var/lib/postgresql/data:Z
    ports:
      - Port:5432
    networks:
      - outline-internal

  outline_minio:
    image: minio/minio
    restart: always
    container_name: outline_minio
    volumes:
      - /volume1/docker/minio:/data:Z
    environment:
      - MINIO_ROOT_USER=minio
      - MINIO_ROOT_PASSWORD=XXXXXXXXXXX
      - MINIO_BROWSER_REDIRECT_URL=https://minio.mydomain.synology.me
   ports:
      - Port:9000
      - Port:9001
   command: "server /data --console-address :9001"
    networks:
      - nginxproxy

  outline:
    image: outlinewiki/outline
    user: root
    restart: always
    container_name: outline
    command: sh -c "yarn db:migrate --env=production-ssl-disabled && yarn start --env=production-ssl-disabled"
    environment:
      - PGSSLMODE=disable
      - SECRET_KEY=random 32 hex numbers
      - UTILS_SECRET=random 32 hex numbers
      - DATABASE_URL=postgres://outline:password@NASIP:Port/outline
      - REDIS_URL=redis://NASIP:Port
      - URL=https://wiki.mydomain.synology.me
      - PORT=3000
      - AWS_ACCESS_KEY_ID=minio
      - AWS_REGION=us-east-2
      - AWS_SECRET_ACCESS_KEY=MinioLoginPW
      - AWS_S3_UPLOAD_BUCKET_URL=https://mini.mydomain.synology.me
      - AWS_S3_UPLOAD_BUCKET_NAME=weekee
      - AWS_S3_UPLOAD_MAX_SIZE=26214400
      - AWS_S3_FORCE_PATH_STYLE=true
      - AWS_S3_ACL=private
      - FORCE_HTTPS=false
      - SLACK_CLIENT_ID=XXXXXXXXX.XXXXXXXXX
      - SLACK_CLIENT_SECRET=XXXXXXXXXXXXXXXXXXXXX
      - SLACK_VERIFICATION_TOKEN=XXXXXXXXXXXXXXXXXX
      - SLACK_APP_ID=XXXXXXXXXXXXXX
      - SLACK_MESSAGE_ACTIONS=false
      - SMTP_HOST=smtp.gmail.com
      - SMTP_PORT=587
      - SMTP_USERNAME=XXXXX@gmail.com
      - SMTP_PASSWORD=XXXXXXXXXX
      - SMTP_FROM_EMAIL=XXXXX@gmail.com
      - SMTP_REPLY_EMAIL=XXXXXX@gmail.com
      - SMTP_TLS_CIPHERS=TLSv1.2
      - SMTP_SECURE=false
      - OIDC_CLIENT_ID=outline
      - OIDC_CLIENT_SECRET=XXXXXXXXXXXX
      - OIDC_AUTH_URI=https://keycloak.mydomain.synology.me/auth/realms/XXX/protocol/openid-connect/auth
      - OIDC_TOKEN_URI=https://key.mydomain.synology.me/auth/realms/XXX/protocol/openid-connect/token
      - OIDC_USERINFO_URI=https://key.mydomain.synology.me/auth/realms/XXX/protocol/openid-connect/userinfo
      - OIDC_USERNAME_CLAIM=XXXXXXXXX@gmail.com
      - OIDC_DISPLAY_NAME=KeyCloak
      - OIDC_SCOPES=openid profile email
    ports:
      - Port:3000
    networks:
      - nginxproxy

networks:
  outline-internal:
  nginxproxy:
    external: true`

Other Details:
-I use Nginx Proxy Manager to route Outline and Minio via reverse proxy

-------

**To Reproduce**

**Expected behavior**

**Screenshots**
If applicable, add screenshots or videos to help explain your problem.

**Outline (please complete the following information):**
- Install: [getoutline.com or self hosted]
- Version: [commit sha if self hosted]

**Desktop (please complete the following information):**

**Mobile (please complete the following information):**

**Comment by nnsee** [comment]
When you open up the Network tab in your browser's dev tools and try to upload, do you see any response to the upload request to your bucket (hopefully with a somewhat descriptive error message)?

**Reply by MetaBliss** [reply]
Thanks @nnsee  I didnt realize I received a response so apologies for the delay. 

Here's a snapshot of my network tab in the browser. Ive redacted my outline and minio domain names but kept the placeholder there so you can see the origin of the network activity. 

What I see is that with the Minio domain I'm getting both CORS failed and NS_ERROR_DOM_BAD_URI and NS_ERROR_NET_INTERRUPT. I'm hoping to figure out how to solve this. I got Outline running its just the problem I'm facing with images/

<img width="2053" alt="Screen Shot 2023-04-07 at 10 41 11 AM" src="https://user-images.githubusercontent.com/84720384/230630613-2e6b3e81-32b0-4d80-a789-9a431f498cb6.png">

**Reply by N4v41** [reply]
into this case it looks like a cors problem, the bucket has the cors correctly configured ?

**Comment by johncmunson** [comment]
I have the exact same problem! Literally everything works.... except for image uploads. Nothing is showing up in the docker container logs, everything appears normal. The network request fails, but it doesn't give me anything useful to really go off of.

<img width="622" alt="image" src="https://github.com/outline/outline/assets/19480078/30ccdf5e-2ce9-42c6-8c0e-59fca258a310">

<img width="728" alt="image" src="https://github.com/outline/outline/assets/19480078/3aa39f10-292b-42ac-a8f6-72182ef5330c">

<img width="483" alt="image" src="https://github.com/outline/outline/assets/19480078/657f3e68-8dbc-433b-a3bf-98c4f067f064">

**Comment by routerino** [comment]
It's a problem with a bug in minio, buckets names cannot also be included in the subdomain.
